sections/discussion/discussion-chapter.tex

```diff
@@ -1,9 +1,153 @@
+
+\subsubsection{Timestamp as a context variable}
+A timestamp is often the only available context variable in larger datasets that can be extracted and used.
+However, an assumption made when doing so is that the rating was given by the user at the time of finishing consumption of the item, if we wish to use it for recommendations.
+For example, if a user just signed up for a service in the summer and starts rating Christmas movies, the system might assume that the user has a preference for watching Christmas movies in the summer.
+To combat this, the collection of data will have to keep track of both when the rating was made, and when the item was consumed.
+This information is not available in any datasets that we are aware of, meaning that the assumption that a rating is made soon after consumption is required.
+As described in \autoref{subsec:desc-of-datasets}, \cite{COMODA2013} defines that some of the most influential context variables are the dominant emotions during the watching of the movie as well as the final emotion felt.
+As such, we can infer that ratings provided shortly after watching do a better job of reflecting the contextual influence on the rating.
+However, this cannot be guaranteed as some users might rate a batch of items in quick succession, or even rate a single item after a substantial amount of time has passed since consumption.
+This means the data could lose some of the contextual impact if timestamp is the only context variable that is available or inferable.
+\autoref{tbl:contextprotocol} defines the context variables used for the experiments, in which it is shown that the MovieLens dataset only has a timestamp for context, meaning this is at least a factor for the results on this dataset. 
+
+\subsubsection{The performance of graph based methods} 
+The results were shortly examined in \autoref{subsec:desc-of-datasets}, where it was noted that the graph based methods generally did not perform well.
+For precision and F1, only NGCF achieved comparable results on a single dataset, which was LDOS-CoMoDa.
+IS-UserBased and KGAT generally showed good performance on recall.
+In terms of frontloading the top-$N$ list, KGAT, NGCF and IS-UserBased show the best results on MovieLens, LDOS-CoMoDa and Yahoo respectively.
+While this does not indicate that the graph-based methods overall outperform the other methods, they show decent performance in terms of frontloading items.
+A reason for the poorer performance of graph based methods might be the sizes of the datasets.
+We suspect this because the datasets that were used for the development of these methods are generally a lot larger.
+For example, the smallest dataset used in the \textit{LightGCN}\cite{LightGCN} method has about a million interactions.
+Therefore it could be interesting to run the experiment on a larger sized dataset to see if the graph-based methods achieve better performance.
```

Comment:
```suggestion
Therefore it could be interesting to run the experiment on a larger dataset to see if the graph-based methods achieve better performance.
```